contracts/src/v0.8/dev/vrf/testhelpers/VRFV2PlusWrapperLoadTestConsumer.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import "../VRFV2PlusWrapperConsumerBase.sol";
+import "../../../shared/access/ConfirmedOwner.sol";
+import "../../../ChainSpecificUtil.sol";
+
+contract VRFV2PlusWrapperLoadTestConsumer is VRFV2PlusWrapperConsumerBase, ConfirmedOwner {
+  uint256 public s_responseCount;
+  uint256 public s_requestCount;
+  uint256 public s_averageFulfillmentInMillions = 0; // in millions for better precision
+  uint256 public s_slowestFulfillment = 0;
+  uint256 public s_fastestFulfillment = 999;
+  uint256 public s_lastRequestId;
+  mapping(uint256 => uint256) requestHeights; // requestIds to block number when rand request was made
+
+  event WrappedRequestFulfilled(uint256 requestId, uint256[] randomWords, uint256 payment);
+  event WrapperRequestMade(uint256 indexed requestId, uint256 paid);
+
+  struct RequestStatus {
+    uint256 paid;
+    bool fulfilled;
+    uint256[] randomWords;
+    uint requestTimestamp;
+    uint fulfilmentTimestamp;
+    uint256 requestBlockNumber;
+    uint256 fulfilmentBlockNumber;
+    bool native;
+  }
+
+  mapping(uint256 => RequestStatus) /* requestId */ /* requestStatus */ public s_requests;
+
+  constructor(
+    address _link,
+    address _vrfV2PlusWrapper
+  ) ConfirmedOwner(msg.sender) VRFV2PlusWrapperConsumerBase(_link, _vrfV2PlusWrapper) {}
+
+  function makeRequests(
+    uint32 _callbackGasLimit,
+    uint16 _requestConfirmations,
+    uint32 _numWords,
+    uint16 _requestCount
+  ) external onlyOwner {
+    for (uint16 i = 0; i < _requestCount; i++) {
+      uint256 requestId = requestRandomness(_callbackGasLimit, _requestConfirmations, _numWords);
+      s_lastRequestId = requestId;
+
+      uint256 requestBlockNumber = ChainSpecificUtil.getBlockNumber();
+      uint256 paid = VRF_V2_PLUS_WRAPPER.calculateRequestPrice(_callbackGasLimit);
+      s_requests[requestId] = RequestStatus({
+        paid: paid,
+        fulfilled: false,
+        randomWords: new uint256[](0),
+        requestTimestamp: block.timestamp,
+        requestBlockNumber: requestBlockNumber,
+        fulfilmentTimestamp: 0,
+        fulfilmentBlockNumber: 0,
+        native: false
+      });
+      s_requestCount++;
+      requestHeights[requestId] = requestBlockNumber;
+      emit WrapperRequestMade(requestId, paid);
+    }
+  }
+
+  function makeRequestsNative(
+    uint32 _callbackGasLimit,
+    uint16 _requestConfirmations,
+    uint32 _numWords,
+    uint16 _requestCount
+  ) external onlyOwner {
+    for (uint16 i = 0; i < _requestCount; i++) {
+      uint256 requestId = requestRandomnessPayInNative(_callbackGasLimit, _requestConfirmations, _numWords);
```

Comment:
perhaps we should consider returning payment amount along with requestID for this method.

Reply:
do we really need that here? otherwise, I will need to return array of request ids and we are already setting this in [s_requests[requestId] = RequestStatus({})](https://github.com/smartcontractkit/chainlink/pull/10721/files#diff-9decad0baea2fea820ff4f0f745d7f26b1e3f1f63359437e67ebc3a469cfa835R78)